### PR TITLE
DS-2929 Bump Mocha version

### PR DESCRIPTION
**Pull request by ildomar-grings**
## Overview

[DS-2929](https://loyaltynz.atlassian.net/browse/DS-2929)

`Minimist <=1.2.5 is vulnerable to Prototype Pollution via file index.js, function setKey() (lines 69-95).`

After update:
```
ildomarg@LNZM-NK0W7K363C /Users/ildomarg/dev/repos/alchemy-ether % npm list minimist
alchemy-ether@1.0.1 /Users/ildomarg/dev/LoyaltyNZ/alchemy-ether
└─┬ mocha@7.2.0
  └─┬ mkdirp@0.5.5
    └── minimist@1.2.8
```

## What is the change?
- Bump `mocha` version to fix critical vulnerabilities
- Bump `coffeescript`, because of `mocha`
- Also, bump minor versions of:
  - `msgpack`
  - `node-uuid`

## Notes

[DS-2929]: https://loyaltynz.atlassian.net/browse/DS-2929?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ildomar-grings** [comment]
retest this please

File: test/service_spec.coffee [hunk 1]
```diff
@@ -577,25 +584,25 @@ describe 'Service', ->
         good_service.start().delay(50) # should get the message that was lost after a bit
       )
       .then( ->
-        expect(recieved_first_time).to.equal true
-        expect(recieved_second_time).to.equal false
+        expect(received_first_time).to.equal true
+        expect(received_second_time).to.equal true # It was a false negative before, because the service was started
```
Comment (davidoram) [inline test/service_spec.coffee:588]: I don't understand why this has changed?

If the message was processed by `dead_service`, and that errors, is it  also sent to `good_service`?

**Review by davidoram — COMMENTED** [review]
Great work updating all the old code.  I just have one question around a test that changed

**Review by davidoram — APPROVED** [review]
LGTM

File: test/service_spec.coffee (hunk 1 above)
Comment (davidoram) [reply test/service_spec.coffee:588]: As discussed the incoming message sits on the queue, and is processed by the second service thats started - see next test case for an example

**Pull request merged**